apps/studio/src/pages/home/library/Marketplace/ConnectWallet.tsx

```diff
@@ -0,0 +1,46 @@
+import { Stack, Typography } from "@mui/material";
+import { Alert, Button } from "@newm-web/elements";
+import theme from "@newm-web/theme";
+import { useConnectWallet } from "@newm.io/cardano-dapp-wallet-connector";
+import { useAppDispatch } from "../../../../common";
+import { setIsConnectWalletModalOpen } from "../../../../modules/ui";
+
+export const ConnectWallet = () => {
+  const dispatch = useAppDispatch();
+  const { wallet, disconnect } = useConnectWallet();
+
+  const handleConnectWallet = () => {
+    if (wallet) {
+      disconnect();
+    }
+
+    dispatch(setIsConnectWalletModalOpen(true));
+  };
+
+  return (
+    <Stack>
+      <Alert>
+        <Typography
+          color={ theme.colors.blue }
+          fontWeight={ 500 }
+          variant="subtitle1"
+        >
+          { wallet
```

Comment:
Once the wallet was connected I had to either refresh the page or switch back and forth between tabs to refresh the component, otherwise it just shows that my wallet doesn't have stream tokens which wasn't valid. Maybe pass a state down from `MarketplaceTab` to allow for a rerender of the Tab?